doc/design/OrionMongoSink.md

```diff
@@ -0,0 +1,110 @@
+#OrionMongoSink
+##Functionality
+`com.iot.telefonica.cygnus.sinks.OrionMongoSink`, or simply `OrionMongosink` is a sink designed to persist NGSI-like context data events within a MongoDB server. Usually, such a context data is notified by a [Orion Context Broker](https://github.com/telefonicaid/fiware-orion) instance, but could be any other system speaking the <i>NGSI language</i>.
+
+Independently of the data generator, NGSI context data is always transformed into internal Flume events at Cygnus sources thanks to `com.iot.telefonica.cygnus.handlers.OrionRestHandler`; click [here](from_ngsi_events_to_flume_events.md) to see the details. In the end, the information within these Flume events must be mapped into specific MongoDB data structures.
+
+###Mapping to MongoDB data structures
+MongoDB organizes the data in databases that contain collections of Json documents. Such organization is exploited by `OrionMongoSink` each time a Flume event is taken, by performing the following workflow:
+
+1. The bytes within the event's body are parsed and a `NotifyContextRequest` object container is created.
+2. A database called as the `fiware-service` header value within the event is created (if not existing yet).
+3. The context responses/entities within the container are iterated, and a collection is created (if not yet existing) for each unit data. The unit data depends on the chosen data model (see the configuration section):
+    * <i>collection-per-service-path</i>: the collection is called as the `fiware-servicePath` header value whitin the event.
+    * <i>collection-per-entity</i>: the collection is called as the concatenation of the `fiware-servicePath`_`destination` headers values within the event.
```

Comment:
Have you take into account the namesapce limit that MongoDB 2.6 has (http://docs.mongodb.org/v2.6/reference/limits/#namespaces)?

>     The limitation on the number of namespaces is the size of the namespace file divided by 628. A 16 megabyte namespace file can support approximately 24,000 namespaces. Each collection and index is a namespace.

Appart of the _id index and considering 1-2 additional index to optimize queries, that would be 4 namespaces per entity, thus a limit of around 6,000. Taking into account that "small" deployments like SmartSantander has around 1,000 (and the complete entitiese set in orion.lab ~50,000 entities) I think that it is too limiting.

Interestingly, Mongo DB 3.0 don't have such limitation if WireTiger storage backend is used (http://docs.mongodb.org/v3.0/reference/limits/#namespaces):

> The WiredTiger storage engine is not subject to this limitation.